cores/esp8266/MD5Builder.cpp

```diff
@@ -23,6 +23,47 @@ void MD5Builder::addHexString(const char * data){
   free(tmp);
 }
 
+bool MD5Builder::addStream(Stream & stream, const int total_len) {
+    const int buf_size = 512;
+    int bytesleft = total_len;
+    int bytestoread;
+    uint8_t * buf = (uint8_t*) malloc(buf_size);
+    if(buf) {
+        while((stream.available() > -1) && (bytesleft > 0)) {
+
+            // get available data size
+            int sizeAvailable = stream.available();
+            if(sizeAvailable) {
+                int readBytes = sizeAvailable;
+
+                // read only the asked bytes
+                if(readBytes > bytesleft) {
+                    readBytes = bytesleft ;
+                }
+
+                // not read more the buffer can handle
+                if(readBytes > buf_size) {
+                    readBytes = buf_size;
+                }
+
+                // read data  
+                int bytesread = stream.readBytes(buf, readBytes);
+                bytesleft -= bytesread;
```

Comment:
Also, here, I think bytesread should be subtracted after the check.. and handle the case where the read is short.